blog/2024-04-23-dlc-channels-demystified.md

```diff
@@ -0,0 +1,317 @@
+    A fee reserve is included in the DLC channel, but it's not a reliable solution.
+
+2.  Users must be able to splice in and out of DLC channels
+
+    *Splicing in* is a term used to describe the process of adding funds to a channel by way of a single on-chain transaction.
+    *Splicing out* is used to describe a similar process, where funds are extracted from a channel.
+
+    These processes can be applied to DLC channels to.
+    Splicing in and out is valuable because it reduces the number of transactions needed to resize a DLC channel.
+
+    10101 users commonly ask to add funds to their DLC channel.
+    At the moment we tell them to close their DLC channel and open a bigger one.
+    Splicing in would let them do so in a single, cheaper step.
+
+3.  Speeding up the DLC protocol
+
+    The 10101 app noticeably slows down when you open a channel, open a new position or resize an existing position.
+    Some of this is inherent to using DLC channels: a lot of signatures and adaptor signatures need to be generated, exchanged and verified.
+    But we are always looking for ways to make this process faster.
+
+    The primary maintainer of the [DLC library](https://github.com/p2pderivatives/rust-dlc) we use and co-maintain, recently came to us with promising benchmarks after replacing ECDSA adaptor signatures with Schnorr adaptor signatures in the on-chain DLC protocol.
+    We expect that by applying the same changes to the DLC channel protocol we will be able to speed up trading in 10101 considerably.
+
+Closing thoughts
+----------------
+
+We have discussed how 10101 uses DLC channels to provide self-custodial and counterparty-risk free trading.
+From opening your first position and DLC channel, to closing your DLC channel with the collaboration of the 10101 coordinator, or with the unbeknownst help of the oracle.
+
+Throughout, we have highlighted the ways in which DLC channels offer a more secure alternative to central exchanges.
+We have also been clear about where problems might arise, and how a full-fledged 10101 will deal with those limitations.
+
+If you have any questions on this or any other topics, do not hesitate to reach us on [Twitter](https://twitter.com/get10101) or [Telegram](https://t.me/get10101/1).
+
+
+Footnotes
+=========
+
+[^1]: Simply put, a perpetual swap or CFD is a contract where one party (the long party) gets paid if the price of a security goes up, and their counterparty (the short party) gets paid if the price of that security goes down.
+
+[^2]: An off-chain transaction is a transaction that hasn't been published yet.
+An off-chain transaction may never be published, as it may be invalidated and replaced by another one.
+
+[^3]: We are not limited to BTCUSD or perpetual futures, but we are focusing on this at the moment.
+
+[^4]: TODO: Link to Richie's blog post.
+
+[^5]: A nonce is a large, secret number that must be used only once in cryptographic schemes.
```

Comment:
Interesting, typically a nonce doesn't have to be secret, but I guess this is different here given the structure of the adapter signatures.

Reply:
This is not specific to adaptor signatures. If you reveal the nonce used to generate a Schnorr signature, you leak your secret key:

For a secret key `x` and a message `M`.

```
gen -> k
K = kG
e = H(K || M)
s = k - xe

sig = (s, e)
```

If you share the signature with someone and then reveal `k`:

```
(s, e) = sig
s = k - xe
s - k = k - xe - k
s - k = xe
(s - k) / e = xe / e
(s - k) / e = x
```